Doesn't build on darwin-arm64 (Apple Silicon)

**Describe the bug**
<!-- What the bug is, how to reproduce, and the expected behavior. -->
From an "Apple Silicon" Mac, run `node-gyp rebuild` from a clone of 3df0940aacef23529826799e18f259638c0b84b4 and observe the following:
<details>
<summary>Click here for `node-gyp rebuild` output</summary>

```
gyp info it worked if it ends with ok
gyp info using node-gyp@8.0.0
gyp info using node@16.2.0 | darwin | arm64
gyp info find Python using Python version 3.9.1 found at "/Users/ssw/.pyenv/versions/3.9.1/bin/python3"
(node:39813) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
gyp info spawn /Users/ssw/.pyenv/versions/3.9.1/bin/python3
gyp info spawn args [
gyp info spawn args   '/Users/ssw/.nodenv/versions/16.2.0/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/ssw/code/dd-trace-js/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/ssw/.nodenv/versions/16.2.0/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/ssw/Library/Caches/node-gyp/16.2.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/ssw/Library/Caches/node-gyp/16.2.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/ssw/.nodenv/versions/16.2.0/lib/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/ssw/Library/Caches/node-gyp/16.2.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/ssw/code/dd-trace-js',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/metrics/packages/dd-trace/src/native/metrics/Collector.o
In file included from ../packages/dd-trace/src/native/metrics/Collector.cpp:1:
In file included from ../packages/dd-trace/src/native/metrics/Collector.hpp:6:
In file included from ../packages/dd-trace/src/native/metrics/Object.hpp:3:
In file included from ../node_modules/nan/nan.h:56:
In file included from /Users/ssw/Library/Caches/node-gyp/16.2.0/include/node/node.h:63:
In file included from /Users/ssw/Library/Caches/node-gyp/16.2.0/include/node/v8.h:30:
/Users/ssw/Library/Caches/node-gyp/16.2.0/include/node/v8-internal.h:452:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
            !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                ~~~~~^~~~~~~~~~~
                                     remove_cv
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:776:50: note: 'remove_cv' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                 ^
1 error generated.
make: *** [Release/obj.target/metrics/packages/dd-trace/src/native/metrics/Collector.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/ssw/.nodenv/versions/16.2.0/lib/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Darwin 20.4.0
gyp ERR! command "/Users/ssw/.nodenv/versions/16.2.0/bin/node" "/Users/ssw/.nodenv/versions/16.2.0/bin/node-gyp" "rebuild"
gyp ERR! cwd /Users/ssw/code/dd-trace-js
gyp ERR! node -v v16.2.0
gyp ERR! node-gyp -v v8.0.0
gyp ERR! not ok 
```

</details>

Seems like the most relevant bit is
```
  CXX(target) Release/obj.target/metrics/packages/dd-trace/src/native/metrics/Collector.o
In file included from ../packages/dd-trace/src/native/metrics/Collector.cpp:1:
In file included from ../packages/dd-trace/src/native/metrics/Collector.hpp:6:
In file included from ../packages/dd-trace/src/native/metrics/Object.hpp:3:
In file included from ../node_modules/nan/nan.h:56:
In file included from /Users/ssw/Library/Caches/node-gyp/16.2.0/include/node/node.h:63:
In file included from /Users/ssw/Library/Caches/node-gyp/16.2.0/include/node/v8.h:30:
/Users/ssw/Library/Caches/node-gyp/16.2.0/include/node/v8-internal.h:452:38: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
            !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
                                ~~~~~^~~~~~~~~~~
                                     remove_cv
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/type_traits:776:50: note: 'remove_cv' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                 ^
1 error generated.
```

I've got nan@2.14.2 installed, which is its current release. Not sure how to discern who's at fault here; I'm new to C++ addons.

**Environment**

* **Operation system:** macOS 11.3 (Big Sur) / Darwin 20.4.0, on arm64 aka **Apple Silicon** aka M1
* **Node version:** 16.2.0
* **Tracer version:** n/a
* **Agent version:** n/a

Also using current release of node-gyp, 8.0.0.

--- a/binding.gyp
+++ b/binding.gyp
@@ -18,7 +18,7 @@
       "<!(node -e \"require('nan')\")"
     ],
     "xcode_settings": {
-      "MACOSX_DEPLOYMENT_TARGET": "10.9",
+      "MACOSX_DEPLOYMENT_TARGET": "10.10",
       "OTHER_CFLAGS": [
         "-std=c++14",
         "-stdlib=libc++",

--- a/binding.gyp
+++ b/binding.gyp
@@ -20,7 +20,7 @@
     "xcode_settings": {
       "MACOSX_DEPLOYMENT_TARGET": "10.9",
       "OTHER_CFLAGS": [
-        "-std=c++11",
+        "-std=c++14",
         "-stdlib=libc++",
         "-Wall",
         "-Werror"

